### PR TITLE
Put VB compiler error IDs in H1 headings for SEO

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #21060 .

(Lots of whitespace trimming - please hide whitespace changes in diff.)

RegEx used:

Find: `# (.*\n)((.|\n)*\*Error ID:\*\* (BC[0-9]{5}?)\s*\n)`
Replace: `# $4: $1$2`

### CODE REVIEW

File: docs/visual-basic/toc.yml [hunk 1]
```diff
@@ -875,14 +875,17 @@
     - name: XML Processing Instruction Literal
       href: language-reference/xml-literals/xml-processing-instruction-literal.md
   - name: Error Messages
-    href: language-reference/error-messages/index.md
     items:
+    - name: Overview
+      href: language-reference/error-messages/index.md
+    - name: BC30663
```
Comment (gewarren) [inline docs/visual-basic/toc.yml:881]: These are the only two that I replaced error messages with error IDs in the TOC so far.

File: docs/visual-basic/language-reference/error-messages/custom-modifier-is-not-valid-on-events-declared-without-explicit-delegate-types.md [hunk 2]
```diff
@@ -1,41 +1,41 @@
...
-  
-     [!code-vb[VbVbalrEventError#19](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrEventError/VB/VbVbalrEventError.vb#19)]  
-  
-## Example  
-
- This example declares a `Custom Event` and specifies the required `As` clause with a delegate type.  
-  
- [!code-vb[VbVbalrEventError#2](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrEventError/VB/VbVbalrEventError.vb#2)]  
-  
+# : BC31122'Custom' modifier is not valid on events declared without explicit delegate types
```
Comment (Youssef1313) [inline docs/visual-basic/language-reference/error-messages/custom-modifier-is-not-valid-on-events-declared-without-explicit-delegate-types.md:11]: For some reason the regex didn't work here as expected.

File: docs/visual-basic/language-reference/error-messages/xml-entity-references-are-not-supported.md [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
   - "bc31180"
-helpviewer_keywords: 
+helpviewer_keywords:
   - "BC31180"
 ms.assetid: 2a393327-d8e2-4187-85b1-642b4f53b4ae
 ---
-# XML entity references are not supported
+# BC31180: XML entity references are not supported
+
+An entity reference (for example, `&copy;`) that is not defined in the XML 1.0 specification is included as a value for an XML literal. Only `&`, `"`, `<`, `>`, and `'` XML entity references are supported in XML literals.
```
Comment (Youssef1313) [inline docs/visual-basic/language-reference/error-messages/xml-entity-references-are-not-supported.md:13]: This changes the rendering. Not sure if this was intended.

File: docs/visual-basic/language-reference/error-messages/custom-modifier-is-not-valid-on-events-declared-without-explicit-delegate-types.md (hunk 2 above)
Comment (gewarren) [reply docs/visual-basic/language-reference/error-messages/custom-modifier-is-not-valid-on-events-declared-without-explicit-delegate-types.md:11]: Some of them I did manually before realizing how tedious it was. I'm sure this was one of those!

File: docs/visual-basic/language-reference/error-messages/xml-entity-references-are-not-supported.md (hunk 3 above)
Comment (gewarren) [reply docs/visual-basic/language-reference/error-messages/xml-entity-references-are-not-supported.md:13]: Good catch! @IEvangelist , is this something that the smart quote replacement tool might have done? It's not rendering as a copyright symbol: https://review.docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/error-messages/xml-entity-references-are-not-supported?branch=pr-en-us-21061. The tool should probably not update anything in code tags.

Comment (gewarren) [inline docs/visual-basic/language-reference/error-messages/xml-entity-references-are-not-supported.md:13]:
```suggestion
An entity reference (for example, `©`) that is not defined in the XML 1.0 specification is included as a value for an XML literal. Only `&`, `"`, `<`, `>`, and `'` XML entity references are supported in XML literals.
```

Comment (IEvangelist) [reply docs/visual-basic/language-reference/error-messages/xml-entity-references-are-not-supported.md]: Yes, you're correct. The smart quotes will replace that when enabled... and it does still need to have that bug addressed where it replaces things in code that it shouldn't.

File: docs/visual-basic/language-reference/error-messages/index.md [hunk 4]
```diff
@@ -1,35 +1,36 @@
...
- By resolving warnings from the Visual Basic compiler, you might be able to write code that runs faster and has fewer bugs. These warnings identify code that may cause errors when the application is run. For example, the compiler warns you if you try to invoke a member of an unassigned object variable, return from a function without setting the return value, or execute a `Try` block with errors in the logic to catch exceptions. For more information about warnings, including how to turn them on and off, see [Configuring Warnings in Visual Basic](/visualstudio/ide/configuring-warnings-in-visual-basic).
+# Error messages in Visual Basic
+
+When you write, compile, or run a Visual Basic application, the following types of errors can occur:
+
+- Design-time errors, which occur when you write an application in Visual Studio.
+
+- Compile-time errors, which occur when you compile an application in Visual Studio or at a command prompt.
+
+- Run-time errors, which occur when you run an application in Visual Studio or as a stand-alone executable file.
```
Comment (Youssef1313) [inline docs/visual-basic/language-reference/error-messages/index.md:20]: Unrelated to the PR.

But this needs to be updated in my opinion.

1. None of these are really tied to Visual Studio.
2. I don't get what design-time errors are? did it means logical errors? in that case the description isn't correct?

Comment (Youssef1313) [reply docs/visual-basic/language-reference/error-messages/index.md:20]: The next headers only describe Run-time errors and Compile-time errors.

File: docs/visual-basic/language-reference/error-messages/line-statements-are-no-longer-supported-visual-basic-compiler-error.md [hunk 5]
```diff
@@ -1,25 +1,25 @@
...
+f1_keywords:
   - "bc30830"
   - "vbc30830"
-helpviewer_keywords: 
+helpviewer_keywords:
   - "BC30830"
 ms.assetid: 4734bc1d-882e-4555-b498-1f1ec0399d16
 ---
-# 'Line' statements are no longer supported (Visual Basic Compiler Error)
+# BC30830: 'Line' statements are no longer supported (Visual Basic Compiler Error)
```
Comment (Youssef1313) [inline docs/visual-basic/language-reference/error-messages/line-statements-are-no-longer-supported-visual-basic-compiler-error.md:11]:
```suggestion
# BC30830: 'Line' statements are no longer supported
```

**Review by BillWagner — APPROVED** [review]
Thanks for these updates @gewarren 

This LGTM, and you can :shipit: when ready.

**Pull request merged**